test/apps/iodemo/io_demo.cc

```diff
@@ -551,8 +549,8 @@ class P2pDemoCommon : public UcxContext {
                 free(_buffer);
                 break;
             default:
-                /* Unreachable - would fail in ctor */
-                abort();
+                /* No operation - dummy buffer */
```

Comment:
maybe add `UCS_MEMORY_TYPE_EXTERNAL`?

Reply:
LAST already means an "undefined" value - IMO better not add UCS enum value which makes sense only for io_demo test.. better even add another bool field

Reply:
right, I missed that it's ' UCS_`..
> WDYT about checking _buffer==NULL?

lets align with `operator[]`, I would test `_extra_buf` and optionally `_buffer==NULL`

Reply:
discussed offline with @evgeny-leksikov  - to test _buffer==NULL